src/content_cao.cpp

```diff
@@ -716,12 +716,26 @@ void GenericCAO::updateLightNoCheck(u8 light_at_pos)
 	}
 }
 
-v3s16 GenericCAO::getLightPosition()
+m4x4f GenericCAO::getTransform()
 {
-	if (m_is_player)
-		return floatToInt(m_position + v3f(0, 0.5 * BS, 0), BS);
+	scene::ISceneNode *node = getSceneNode();
+	if (node)
+		return m4x4f(node->getAbsoluteTransformation());
```

Comment:
Why is this initialization required? The value is copied automatically unless you pass it as reference.